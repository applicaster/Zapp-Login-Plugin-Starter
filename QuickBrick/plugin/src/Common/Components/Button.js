import React from 'react';
import {
  Text,
  ImageBackground
} from 'react-native';
import { Focusable } from '@applicaster/zapp-react-native-ui-components/Components/Focusable';


export default function Button(
  {
    label = '',
    callback,
    buttonStyle = {},
    textStyle = {},
    backgroundButtonUri = '',
    backgroundButtonUriActive = ''
  }
) {
  return (
    <Focusable id={`tv-button-${label}`} onPress={callback}>
      {
        (focused) => {
          return (
            <ImageBackground
              source={{ uri: focused ? backgroundButtonUriActive : backgroundButtonUri }}
              style={focused ? { ...buttonStyle, ...styles.focused } : buttonStyle}
            >
              <Text
                style={focused ? { ...textStyle, color: '#5F5F5F' } : textStyle}
                adjustsFontSizeToFit
              >
                {label}
              </Text>
            </ImageBackground>
          );
        }
      }
    </Focusable>
  );
}

const styles = {
  focused: {
    backgroundColor: 'rgba(255, 255, 255, 0.8)'
  }
};
