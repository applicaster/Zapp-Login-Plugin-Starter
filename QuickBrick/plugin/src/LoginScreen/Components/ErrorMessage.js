import React from 'react';
import { View, Text } from 'react-native';

const ErrorMessage = ({ errorValue = '', customStyles = {} }) => (
  <View style={styles.container}>
    <Text style={customStyles.fieldsEmptyErrorStyle}>
      {errorValue}
    </Text>
  </View>
);

const styles = {
  container: {
    marginTop: 52
  }
};

export default ErrorMessage;
