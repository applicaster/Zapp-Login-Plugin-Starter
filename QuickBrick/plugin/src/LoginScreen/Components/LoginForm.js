import React, {useState} from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  ActivityIndicator
} from 'react-native';
import ButtonComponent from '../../Common/Components/Button';

export default function LoginForm({onLogin}) {

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);

  return (
    <View>
      <TextInput
        value={username}
        onChangeText={(username) => setUsername(username)}
        placeholder={'Username'}
        style={styles.input}
      />
      <TextInput
        value={password}
        onChangeText={(password) => setPassword(password)}
        placeholder={'Password'}
        secureTextEntry={true}
        style={styles.input}
      />
      {
        loading
        ? <ActivityIndicator />
        : <ButtonComponent
            label='Log In'
            callback={onLogin}
            buttonStyle={styles.button}
            textStyle={styles.buttonText}
          />
      }

    </View>
  )
}

const styles = StyleSheet.create({
  input: {
    width: 550,
    height: 80,
    padding: 20,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#979797',
    marginBottom: 20,
    fontSize: 30,
    fontWeight: 'normal'
  },
  button: {
    width: 550,
    height: 90,
    padding: 20,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#979797',
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonText: {
    fontSize: 30,
    fontWeight: 'normal',
    color: '#545A5C'
  }
});
