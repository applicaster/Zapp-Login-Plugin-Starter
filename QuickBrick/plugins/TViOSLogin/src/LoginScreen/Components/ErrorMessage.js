import React from 'react';
import { View, Text } from 'react-native';

const ErrorMessage = ({ errorValue = '', customStyles = {} }) => (
  <View style={styles.container}>
    <Text style={customStyles.fieldsEmptyErrorStyle}>
      {errorValue}
    </Text>
  </View>
);

const styles = {
  container: {
    marginTop: '8%'
  }
};

export default ErrorMessage;
