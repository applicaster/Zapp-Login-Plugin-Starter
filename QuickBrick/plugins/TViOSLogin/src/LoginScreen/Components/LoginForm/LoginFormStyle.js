import { getPluginData } from '../../../Common/Utils';

export default function createStyleSheet(screenData) {
  const {
    username_input_fontcolor: usernameFontColor,
    username_input_fontsize: usernameFontSize,
    password_input_fontcolor: passwordFontColor,
    password_input_fontsize: passwordFontSize,
    login_action_button_fontcolor: loginButtonFontColor,
    login_action_button_fontsize: loginButtonFontSize,
    skip_action_button_fontcolor: skipButtonFontColor,
    skip_action_button_fontsize: skipButtonFontSize
  } = getPluginData(screenData);

  const usernameInputStyle = {
    color: usernameFontColor,
    fontSize: usernameFontSize
  };

  const passwordInputStyle = {
    color: passwordFontColor,
    fontSize: passwordFontSize
  };

  const loginButtonStyle = {
    color: loginButtonFontColor,
    fontSize: loginButtonFontSize
  };

  const skipButtonStyle = {
    color: skipButtonFontColor,
    fontSize: skipButtonFontSize
  };

  return {
    usernameInputStyle,
    passwordInputStyle,
    loginButtonStyle,
    skipButtonStyle
  };
}
