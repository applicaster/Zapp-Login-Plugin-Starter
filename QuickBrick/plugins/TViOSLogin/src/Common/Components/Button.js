import React from 'react';
import {
  Text,
  ImageBackground
} from 'react-native';
import { Focusable } from '@applicaster/zapp-react-native-ui-components/Components/Focusable';


export default function Button(
  {
    label = '',
    onPress,
    buttonStyle = {},
    textStyle = {},
    backgroundButtonUri = '',
    backgroundButtonUriActive = ''
  }
) {
  return (
    <Focusable id={`tv-button-${label}`} onPress={onPress}>
      {
        (focused) => {
          return (
            <ImageBackground
              source={{ uri: focused ? backgroundButtonUriActive : backgroundButtonUri }}
              style={buttonStyle}
            >
              <Text
                style={focused ? { ...textStyle, color: '#5F5F5F' } : textStyle}
                adjustsFontSizeToFit
              >
                {label}
              </Text>
            </ImageBackground>
          );
        }
      }
    </Focusable>
  );
}
