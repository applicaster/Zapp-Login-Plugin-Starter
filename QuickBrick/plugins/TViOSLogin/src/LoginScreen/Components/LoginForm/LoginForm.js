import React from 'react';
import {
  ActivityIndicator,
  View,
  SafeAreaView
} from 'react-native';
import { Formik } from 'formik';
import validationSchema from '../../Utils/validation';
import Button from '../../../Common/Components/Button';
import Input from '../Input';
import createStyleSheet from './LoginFormStyle';
import ASSETS from './LoginFormAssets';


export default function LoginForm(props) {
  const {
    onLogin,
    screenData,
    isLoading,
    handleSkip,
    handleError
  } = props;

  const customStyles = createStyleSheet(screenData);
  const {
    general: {
      username_input_placeholder: usernamePlaceholder = '',
      password_input_placeholder: passwordPlaceholder = '',
      login_action_button_text: loginLabel = '',
      skip_action_button_text: skipLabel = '',
      enable_skip_functionality: skip = false
    }
  } = screenData;

  const handleValidation = async (validateForm, handleSubmit) => {
    try {
      const errors = await validateForm();
      return (Object.keys(errors).length > 0)
        ? handleError(errors.username || errors.password)
        : handleSubmit();
    } catch (err) {
      console.log(err);
    }
  };

  const handleOnLogin = async (values, actions) => {
    const { username, password } = values;
    onLogin(username, password);
  };

  return (
    <Formik
      initialValues={{ username: '', password: '' }}
      onSubmit={(values, actions) => handleOnLogin(values, actions)}
      validationSchema={validationSchema}
    >
      {(
        {
          handleChange,
          values,
          handleSubmit,
          validateForm
        }
      ) => (
        <SafeAreaView>
          <View style={styles.container}>
            <Input
              value={values.username}
              onChangeText={handleChange('username')}
              secureTextEntry={false}
              placeholder={usernamePlaceholder}
              style={{ ...styles.input, ...customStyles.usernameInputStyle }}
              inputAsset={ASSETS.usernameInputBackground}
              inputAssetActive={ASSETS.usernameInputBackgroundActive}
            />
            <Input
              value={values.password}
              onChangeText={handleChange('password')}
              secureTextEntry
              placeholder={passwordPlaceholder}
              style={{ ...styles.input, ...customStyles.passwordInputStyle, marginBottom: 0 }}
              inputAsset={ASSETS.usernameInputBackground}
              inputAssetActive={ASSETS.usernameInputBackgroundActive}
            />
          </View>
          <View style={styles.container}>
            {
              isLoading
                ? <ActivityIndicator size="large" />
                : (
                  <>
                    <Button
                      label={loginLabel}
                      buttonStyle={styles.input}
                      onPress={() => handleValidation(validateForm, handleSubmit)}
                      textStyle={customStyles.loginButtonStyle}
                      backgroundButtonUri={ASSETS.loginButtonBackground}
                      backgroundButtonUriActive={ASSETS.loginButtonBackgroundActive}
                    />
                    {
                      skip
                      && (
                        <Button
                          label={skipLabel}
                          onPress={handleSkip}
                          buttonStyle={styles.input}
                          textStyle={customStyles.skipButtonStyle}
                          backgroundButtonUri={ASSETS.skipButtonBackground}
                          backgroundButtonUriActive={ASSETS.skipButtonBackgroundActive}
                        />
                      )
                    }
                  </>
                )
            }
          </View>
        </SafeAreaView>
      )}
    </Formik>
  );
}

const styles = {
  container: {
    minWidth: '31.8%',
    minHeight: '6%',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: '7%'
  },
  input: {
    minWidth: 610,
    minHeight: 100,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: '1.5%'
  }
};
