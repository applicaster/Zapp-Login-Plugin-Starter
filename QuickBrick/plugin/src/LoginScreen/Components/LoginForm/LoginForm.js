import React, { useState } from 'react';
import {
  ActivityIndicator,
  View,
  SafeAreaView
} from 'react-native';
import { Formik } from 'formik';
import validationSchema from '../../Utils/validation';
import Button from '../../../Common/Components/Button';
import Input from '../Input';
import ErrorMessage from '../ErrorMessage';
import createStyleSheet from './LoginFormStyle';
import ASSETS from './LoginFormAssets';


export default function LoginForm(props) {
  const {
    onLogin,
    screenData,
    error,
    handleSkip
  } = props;

  const [loading, setLoading] = useState(false);

  const customStyles = createStyleSheet(screenData);
  const {
    general: {
      username_input_placeholder: usernamePlaceholder = '',
      password_input_placeholder: passwordPlaceholder = '',
      login_action_button_text: loginLabel = '',
      skip_action_button_text: skipLabel = '',
      enable_skip_functionality: skip = false
    }
  } = screenData;

  const handleOnLogin = async (values, actions) => {
    const { username, password } = values;
    try {
      setLoading(true);
      onLogin(username, password);
    } catch (err) {
      setLoading(false);
      throw err;
    }
  };

  return (
    <Formik
      initialValues={{ username: '', password: '' }}
      onSubmit={(values, actions) => {
        handleOnLogin(values, actions);
      }}
      validationSchema={validationSchema}
    >
      {(
        {
          handleChange,
          values,
          handleSubmit,
          errors,
          touched
        }
      ) => (
        <SafeAreaView>
          <View style={styles.container}>
            <Input
              value={values.username}
              onChangeText={handleChange('username')}
              secureTextEntry={false}
              placeholder={usernamePlaceholder}
              style={{ ...styles.input, ...customStyles.usernameInputStyle }}
              inputAsset={ASSETS.usernameInputBackground}
              inputAssetActive={ASSETS.usernameInputBackgroundActive}
            />
            <Input
              value={values.password}
              onChangeText={handleChange('password')}
              secureTextEntry
              placeholder={passwordPlaceholder}
              style={{ ...styles.input, ...customStyles.passwordInputStyle, marginBottom: 0 }}
              inputAsset={ASSETS.usernameInputBackground}
              inputAssetActive={ASSETS.usernameInputBackgroundActive}
            />
          </View>
          <View style={styles.container}>
            {
              (loading && !error)
                ? <ActivityIndicator size="large" />
                : (
                  <>
                    <Button
                      label={loginLabel}
                      buttonStyle={styles.input}
                      onPress={handleSubmit}
                      textStyle={customStyles.loginButtonStyle}
                      backgroundButtonUri={ASSETS.loginButtonBackground}
                      backgroundButtonUriActive={ASSETS.loginButtonBackgroundActive}
                    />
                    {
                      skip
                      && (
                        <Button
                          label={skipLabel}
                          onPress={handleSkip}
                          buttonStyle={styles.input}
                          textStyle={customStyles.loginButtonStyle}
                          backgroundButtonUri={ASSETS.skipButtonBackground}
                          backgroundButtonUriActive={ASSETS.skipButtonBackgroundActive}
                        />
                      )
                    }
                  </>
                )
            }
            <ErrorMessage
              errorValue={(touched.password && errors.password) || (touched.username && errors.username)}
              customStyles={customStyles}
            />
          </View>
        </SafeAreaView>
      )}
    </Formik>
  );
}

const styles = {
  container: {
    minWidth: '31.8%',
    minHeight: '6%',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: '7%'
  },
  input: {
    minWidth: 610,
    minHeight: 100,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: '1.5%'
  }
};
