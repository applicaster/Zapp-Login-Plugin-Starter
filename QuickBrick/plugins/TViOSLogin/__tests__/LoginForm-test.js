import React from 'react';
import { create, act } from 'react-test-renderer';
import LoginForm from '../src/LoginScreen/Components/LoginForm/LoginForm';
import Button from '../src/Common/Components/Button';
import ErrorMessage from '../src/LoginScreen/Components/ErrorMessage';


describe('LoginForm', () => {
  const screenData = {
    general: {
      enable_skip_functionality: false
    }
  };

  it('renders correctly', () => {
    const error = null;

    const tree = create(<LoginForm screenData={screenData} error={error} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('check skip button displayed', () => {
    screenData.general.enable_skip_functionality = true;
    screenData.general.skip_action_button_text = 'Skip';

    const testRenderer = create(
      <LoginForm screenData={screenData} />
    );
    const testInstance = testRenderer.root;
    const buttons = testInstance.findAllByType(Button);

    expect(buttons).toHaveLength(2);
    expect(buttons[1].props.label).toBe(`${screenData.general.skip_action_button_text}`);
  });

  it('check error displayed on Login button click', async () => {
    const testRenderer = create(
      <LoginForm screenData={screenData} />
    );
    const testInstance = testRenderer.root;
    const loginButton = testInstance.findAllByType(Button)[0];
    loginButton.props.onPress();

    const error = testInstance.findAllByType(ErrorMessage);

    expect(error).toHaveLength(1);
  });

  it('check spinner displayed on submitting form', async () => {
    const rendered = create(
      <LoginForm screenData={screenData} />
    );

    const [usernameInput, passwordInput] = rendered.root.findAllByType('TextInput');
    const [loginButton] = rendered.root.findAllByType(Button);

    usernameInput.value = 'username';
    passwordInput.value = 'password';
    loginButton.props.onPress();
    rendered.update(<LoginForm screenData={screenData} />);

    const spinner = rendered.root.findAllByType('ActivityIndicator');
    expect(spinner).toHaveLength(0);
  });
});
