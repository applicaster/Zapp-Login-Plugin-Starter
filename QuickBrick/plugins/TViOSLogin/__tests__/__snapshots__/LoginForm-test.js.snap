// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "7%",
        "minHeight": "6%",
        "minWidth": "31.8%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "color": undefined,
          "fontSize": NaN,
          "justifyContent": "center",
          "marginBottom": 15,
          "minHeight": 100,
          "minWidth": 610,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "login_username_input_field_asset.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "alignItems": "center",
            "color": undefined,
            "fontSize": NaN,
            "justifyContent": "center",
            "marginBottom": "1.5%",
            "minHeight": 100,
            "minWidth": 610,
            "padding": 20,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "color": undefined,
          "fontSize": NaN,
          "justifyContent": "center",
          "marginBottom": 15,
          "minHeight": 100,
          "minWidth": 610,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "login_username_input_field_asset.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "color": undefined,
            "fontSize": NaN,
            "justifyContent": "center",
            "marginBottom": 0,
            "minHeight": 100,
            "minWidth": 610,
            "padding": 20,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "7%",
        "minHeight": "6%",
        "minWidth": "31.8%",
      }
    }
  >
    <View
      groupId={null}
      itemId={null}
      onLayout={[Function]}
      onPress={[Function]}
      onViewBlur={[Function]}
      onViewFocus={[Function]}
      onViewPress={[Function]}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": "1.5%",
            "minHeight": 100,
            "minWidth": 610,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "login_action_button_asset.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <Text
          adjustsFontSizeToFit={true}
          style={
            Object {
              "color": undefined,
              "fontSize": NaN,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": "8%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": undefined,
            "fontSize": NaN,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
